agent/command/exec.go

```diff
@@ -263,13 +280,41 @@ func (c *subprocessExec) getProc(ctx context.Context, taskID string, logger clie
 	return cmd
 }
 
-func addTempDirs(env map[string]string, dir string) {
-	for _, key := range []string{"TMP", "TMPDIR", "TEMP"} {
-		if _, ok := env[key]; ok {
-			continue
+// getExecutablePath returns the absolute path to the command executable to run.
+// If the executable is available in the default runtime environment's PATH,
+// then that path will be used. Otherwise if it can't find the command in the
+// default PATH locations, the command will fall back to checking the command's
+// PATH environment variable for a matching executable location (if any).
+func (c *subprocessExec) getExecutablePath(logger client.LoggerProducer) (absPath string, err error) {
+	cmdPath := c.Env["PATH"]
+	defaultPath, err := exec.LookPath(c.Binary)
+	if defaultPath != "" {
+		return defaultPath, err
+	}
+
```

Comment:
Perhaps we should an execution log in this function saying we are falling back to the commands path?

Reply:
Good idea, added one.